File: auth-service/src/main/java/com/pinnacle/vendra/auth/service/AuthService.java
package com.pinnacle.vendra.auth.service;

import com.pinnacle.vendra.auth.dto.*;
import com.pinnacle.vendra.common.dto.UserDto;

public interface AuthService {
    UserDto register(RegisterRequest request);
    LoginResponse login(LoginRequest request);
    TokenRefreshResponse refreshToken(String refreshToken);
    void logout(String refreshToken);
    UserDto getUserProfile(String username);
}

File: frontend/src/services/authService.js
import api from './api';

const authService = {
  login: async (username, password) => {
    const response = await api.post('/auth/login', { username, password });
    return response.data.data; // Returns LoginResponse containing accessToken and user details
  },

  register: async (username, email, password) => {
    const response = await api.post('/auth/register', { username, email, password });
    return response.data.data; // Returns UserDto
  },

  logout: async () => {
    const response = await api.post('/auth/logout');
    return response.data;
  },

  getProfile: async () => {
    const response = await api.get('/auth/profile');
    return response.data.data; // Returns UserDto
  },
};

export default authService;
